components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -1427,68 +1429,127 @@ public Subscriber getSubscriber(String subscriberName) throws APIManagementExcep
      * @throws org.wso2.carbon.apimgt.api.APIManagementException
      *          if failed to get SubscribedAPIs
      */
-    public Set<SubscribedAPI> getSubscribedAPIs(Subscriber subscriber,String applicationName)
+    public Set<SubscribedAPI> getSubscribedAPIs(Subscriber subscriber,String applicationName, String groupId)
             throws APIManagementException {
         Set<SubscribedAPI> subscribedAPIs = new LinkedHashSet<SubscribedAPI>();
         Connection connection = null;
         PreparedStatement ps = null;
         ResultSet result = null;
+        String sqlQuery;
 
         try {
             connection = APIMgtDBUtil.getConnection();
 
-            String sqlQuery = "SELECT " +
-                              "   SUBS.SUBSCRIPTION_ID" +
-                              "   ,API.API_PROVIDER AS API_PROVIDER" +
-                              "   ,API.API_NAME AS API_NAME" +
-                              "   ,API.API_VERSION AS API_VERSION" +
-                              "   ,SUBS.TIER_ID AS TIER_ID" +
-                              "   ,APP.APPLICATION_ID AS APP_ID" +
-                              "   ,SUBS.LAST_ACCESSED AS LAST_ACCESSED" +
-                              "   ,SUBS.SUB_STATUS AS SUB_STATUS" +
-                              "   ,SUBS.SUBS_CREATE_STATE AS SUBS_CREATE_STATE" +
-                              "   ,APP.NAME AS APP_NAME " +
-                              "   ,APP.CALLBACK_URL AS CALLBACK_URL " +
-                              "FROM " +
-                              "   AM_SUBSCRIBER SUB," +
-                              "   AM_APPLICATION APP, " +
-                              "   AM_SUBSCRIPTION SUBS, " +
-                              "   AM_API API " +
-                              "WHERE " +
-                              "   SUB.USER_ID = ? " +
-                              "   AND SUB.TENANT_ID = ? " +
-                              "   AND SUB.SUBSCRIBER_ID=APP.SUBSCRIBER_ID " +
-                              "   AND APP.APPLICATION_ID=SUBS.APPLICATION_ID " +
-                              "   AND API.API_ID=SUBS.API_ID" +
-                              "   AND APP.NAME= ? " +
-                              "   AND SUBS.SUBS_CREATE_STATE = '" + APIConstants.SubscriptionCreatedStatus.SUBSCRIBE + "'";
+            if(groupId == null || "".equals(groupId)) {
```

Comment:
I don't think we should maintain two SQL scripts for this. Doing so will make it hard to maintain the SQL script. We can do a conditional check and construct the WHERE clause only.